File: combo/combo.pas
{
	Author:wind1740
	LANG:PASCAL
	PROG:combo
}
var n:longint;
    mapp:array [1..2,1..3] of longint;
	jetzt:array [1..3] of longint;
	flag:array [1..100,1..100,1..100] of boolean;
	sum:longint;

	
procedure con(was,wo:longint);
var i:longint;
begin
	if wo=4 then 
	begin
		if flag[jetzt[1],jetzt[2],jetzt[3]]=false then
		begin
			flag[jetzt[1],jetzt[2],jetzt[3]]:=true;
			inc(sum);
		end;
	end
	else
	begin
		for i:=(-2) to (2) do 
		
			
		begin
			jetzt[wo]:=i+mapp[was,wo];
			while jetzt[wo]>n do 
			begin
				dec(jetzt[wo],n);
			end;
			while jetzt[wo]<=0 do
				inc(jetzt[wo],n);
			con(was,wo+1);
		end;
		
	end;
end;
	
procedure main;
var i,j,k:longint;
begin
	readln (n);
	for i:=1 to n do
		for j:=1 to n do
			for k:=1 to n do
				flag[i,j,k]:=false;
	for i:=1 to 2 do 
		for j:=1 to 3 do
			read (mapp[i,j]);
	con(1,1);
	con(2,1);
	writeln (sum);
end;

begin
	assign (input,'combo.in');
	assign (output,'combo.out');
	reset (input);
	rewrite (output);
	
	main;
	
	close (input);
	close (output);
end.

File: combo/combo.py
'''
	Author:wind1740
	LANG:PYTHON3
	PROG:combo
'''
	
def con(was,wo):
	global jetzt
	if wo==4 : 
		if flag[jetzt[1]][jetzt[2]][jetzt[3]]==False:
			flag[jetzt[1]][jetzt[2]][jetzt[3]]=True
			global sum
			sum+=1
	else :
		for i in range (-2,3):
			
			jetzt[wo]=i+mapp[was][wo-1]
			while jetzt[wo]>n :
				jetzt[wo]-=n
			while jetzt[wo]<=0 :
				jetzt[wo]+=n
			con(was,wo+1)
	
inn=open ("combo.in","r")
out=open ("combo.out","w")
n=int (inn.readline());
flag = [[[False for i in range(101)] for j in range(101)] for k in range(101)]
jetzt=[0 for i in range (5)]
mapp=[[0 for i in range (3)] for j in range (4)]
sum=0
for i in range (1,3) :
	mapp[i]=list (map(int,inn.readline().split()))
con(1,1);
con(2,1);
out.write (str(sum)+'\n');
inn.close()
out.close()
